app/Libraries/Blog.php

```diff
@@ -0,0 +1,280 @@
+<?php namespace App\Libraries;
+
+use App\Entities\Post;
+use App\Exceptions\BlogException;
+use CodeIgniter\Exceptions\PageNotFoundException;
+use Config\Blog as BlogConfig;
+use League\CommonMark\CommonMarkConverter;
+
+/**
+ * Class Blog
+ *
+ * Handles reading and parsing content posts.
+ * Posts are markdown files versioned with the
+ * site repo, with a simple yaml header section
+ * to define meta data.
+ */
+class Blog
+{
+    /**
+     * @var BlogConfig
+     */
+    protected $config;
+
+    public function __construct()
+    {
+        $this->config = config(BlogConfig::class);
+    }
+
+    /**
+     * Gets the posts, sorted by most recent first.
+     * If $category is present, will locate within a
+     * subfolder of that name.
+     *
+     * @param int         $limit
+     * @param int         $offset
+     * @param string|null $category
+     *
+     * @throws BlogException
+     */
+    public function getRecentPosts(int $limit=5, int $offset=0, string $category = null)
+    {
+        $cacheKey = "blog_files_{$offset}_{$limit}_{$category}";
+
+        if (! $posts = cache($cacheKey)) {
+            helper('filesystem');
+
+            if (! is_dir($this->config->contentPath)) {
+                log_message('Blog Content Path is not a valid directory: '. $this->config->contentPath);
+                throw BlogException::forInvalidContent();
+            }
+
+            $files = get_filenames($this->config->contentPath);
+
+            if (! count($files)) {
+                throw BlogException::forInvalidContent();
+            }
+
+            // Don't trust filesystem, order by date.
+            asort($files);
```

Comment:
The updated `get_filenames()` already sorts content: https://github.com/codeigniter4/CodeIgniter4/blob/e8ead2934b004b41a8ec74287f40b15301e62a44/system/Helpers/filesystem_helper.php#L266

This can be removed.